gateway/src/main/java/io/zeebe/gateway/metrics/GatewayMetrics.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.gateway.metrics;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Histogram;
+
+public final class GatewayMetrics {
+
+  private static final Histogram REQUEST_LATENCY =
+      Histogram.build()
+          .namespace("zeebe")
+          .name("gateway_request_latency")
+          .help("Latency of round-trip from gateway to broker")
+          .labelNames("partition", "requestType")
+          .register();
+
+  private static final Counter FAILED_REQUESTS =
+      Counter.build()
+          .namespace("zeebe")
+          .name("gateway_failed_requests")
+          .help("Number of failed requests")
+          .labelNames("partition", "requestType")
+          .register();
+
+  private static final Counter TOTAL_REQUESTS =
+      Counter.build()
+          .namespace("zeebe")
+          .name("gateway_total_requests")
+          .help("Number of requests")
+          .labelNames("partition", "requestType")
+          .register();
+
+  private GatewayMetrics() {}
+
+  public static void registerSuccessfulRequest(
+      final long partition, final String requestType, final long latencyMs) {
+    REQUEST_LATENCY.labels(Long.toString(partition), requestType).observe(latencyMs / 1000f);
+    TOTAL_REQUESTS.labels(Long.toString(partition), requestType).inc();
```

Comment:
TOTAL_REQUESTS should be updated when the request is send, not when the response is received. Isn't it?

Reply:
@MiguelPires What's your opinion on this?

Reply:
Sorry, missed this comment before. Hm, I wouldn't expect a big difference if the rate at which incoming requests come remains steady. But it seems more useful to consider TOTAL_REQUESTS as "all requests that were completed" rather than "all requests that were made including unfinished ones". If you have a lot of requests that haven't completed then I don't think we should take them into account when calculating the failure rate, since it could heavily skew the results

Reply:
:+1: